packages/teams-js/src/public/runtime.ts

```diff
@@ -58,55 +58,21 @@ export interface IRuntime {
   };
 }
 
-export let runtime: IRuntime = {
+/**
+ * @hidden
+ * Constant used to set the runtime configuration
+ * to its uninitialized state.
+ *
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export const _uninitializedRuntime = {
   apiVersion: 1,
-  supports: {
-    appInstallDialog: undefined,
-    barCode: undefined,
-    calendar: undefined,
-    call: undefined,
-    chat: undefined,
-    webStorage: undefined,
-    conversations: undefined,
-    dialog: {
-      bot: undefined,
-      update: undefined,
-    },
-    geoLocation: {
-      map: undefined,
-    },
-    location: undefined,
-    logs: undefined,
-    mail: undefined,
-    meetingRoom: undefined,
-    menus: undefined,
-    monetization: undefined,
-    notifications: undefined,
-    pages: {
-      appButton: undefined,
-      backStack: undefined,
-      config: undefined,
-      currentApp: undefined,
-      fullTrust: undefined,
-      tabs: undefined,
-    },
-    people: undefined,
-    permissions: undefined,
-    profile: undefined,
-    remoteCamera: undefined,
-    search: undefined,
-    sharing: undefined,
-    stageView: undefined,
-    teams: {
-      fullTrust: {
-        joinedTeams: undefined,
-      },
-    },
-    teamsCore: undefined,
-    video: undefined,
-  },
+  supports: {},
 };
 
+export let runtime: IRuntime = _uninitializedRuntime;
```

Comment:
>_uninitializedRuntime [](http://example.com/codeflow?start=31&length=21)

Is there still value in having an uninitializedRuntime? Or should this now be declared as
```
export let runtime: IRuntime | null = null
```

?

Reply:
Seems reasonable. I think I will leave _uninitializedRuntime in place and just change it to null. That way if we come across some reason to make uninitializedRuntime something OTHER than null in the future we won't need to go back in and change a billion files again.

Reply:
Okay, I went a bit down this parth and I think I'm going to leave it as-is for now. Pretty much every isSupported function in our codebase is written like this:

`return runtime.supports.dialog ? (runtime.supports.dialog.bot ? true : false) : false`

If we make runtime nullable every function gets more annoying to write, and making this nullable is (I think) duplicative with the change I've already got to add ensureInitialized to every isSupported function. Since I think ensureInitialized gives more correct errors than failing isSupported and saying the capability isn't supported because runtime doesn't exist / doesn't have a supports member, going to leave it.

If you disagree let me know!